wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -509,9 +510,18 @@ def add_model(self, model):
         mapping = self.mapping_class(model)
 
         # Put mapping
-        self.es.indices.put_mapping(
-            index=self.name, doc_type=mapping.get_document_type(), body=mapping.get_mapping()
-        )
+        try:
+            self.es.indices.put_mapping(
+                index=self.name, doc_type=mapping.get_document_type(), body=mapping.get_mapping()
+            )
+        except RequestError as e:
+            if 'No handler for type [attachment] declared' in str(e.info):
```

Comment:
I think checking errors like this is a little flaky. They could reword it at any time and we wouldn't know to update this check. We need a unit test to tell us that this check does catch the error correctly.